libs/vscode/nx-project-view/src/lib/views/nx-project-tree-view.ts

```diff
@@ -102,22 +94,22 @@ class TreeView extends BaseView {
    * Each entry is added n times.
    * n is determined by the directory depth.
    */
-  private groupByRootPath(projectDefs: {
-    [projectName: string]: ProjectConfiguration;
-  }) {
-    return Object.entries(projectDefs)
+  private groupByRootPath(projectDefs: ProjectDefinition): TreeViewMap {
+    return objectEntries(projectDefs)
       .flatMap((project) => {
         const [projectName, projectDef] = project;
-        const { root } = projectDef;
+        const nodeModulesInstalled = typeof projectDef === 'string';
```

Comment:
Can you maybe provide a comment on why `projectDef` being a string means the node_modules are installed and what this means for the projects view? I think it would help both current and future me understand what's going on here.